packages/core/src/utils/listen.ts

```diff
@@ -0,0 +1,68 @@
+import { ComponentInterface } from '@stencil/core'
+import { addEventListener, removeEventListener } from './helpers'
+import { balBrowser } from './browser'
+
+export interface ListenOptions {
+  target?: ListenTargetOptions
+  capture?: boolean
+  passive?: boolean
+}
+
+export type ListenTargetOptions = 'document' | 'window'
+
+/**
+ * This decorator replaces the @Listen from stencil since
+ * they have issue with connecting and disconnecting web-components.
+ */
+export function ListenTo(eventName: string, opts: ListenOptions = {}) {
```

Comment:
Should we indicate in the function's name that it's not coming from stencil anymore but is part of the design system?

Reply:
see that form the import